### ISSUE TITLE
[Bug]: Exception during scan: File entry could not be inserted 

### ISSUE THREAD

**Issue by jacksbox** [nextcloud/recognize#949]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I copied over new files into my users nextcloud dir (mounted to the host via cifs, not as external storage in nc) and wanted to add them to nextcloud via `files:scan`.
But this always results in following error (with `-v`):
```
Folder	/mario/files
File	/mario/files/test.jpg
File	/mario/files/test.jpg
Exception during scan: File entry could not be inserted but could also not be selected with getId() in order to perform an update. Please try again.
#0 /var/www/html/lib/private/Files/Cache/Scanner.php(300): OC\Files\Cache\Cache->insert('files/test.jpg', Array)
nextcloud/server#1 /var/www/html/lib/private/Files/Cache/Scanner.php(233): OC\Files\Cache\Scanner->addToCache('files/test.jpg', Array, -1)
nextcloud/server#2 /var/www/html/lib/private/Files/Cache/Scanner.php(457): OC\Files\Cache\Scanner->scanFile('files/test.jpg', 3, 107322, false, true, Array)
nextcloud/server#3 /var/www/html/lib/private/Files/Cache/Scanner.php(397): OC\Files\Cache\Scanner->handleChildren('files', true, 3, 107322, true, 0)
nextcloud/server#4 /var/www/html/lib/private/Files/Cache/Scanner.php(400): OC\Files\Cache\Scanner->scanChildren('files', true, 3, 107322, true, -1)
nextcloud/server#5 /var/www/html/lib/private/Files/Cache/Scanner.php(347): OC\Files\Cache\Scanner->scanChildren('', true, 3, 107314, true, 725)
nextcloud/server#6 /var/www/html/lib/private/Files/Utils/Scanner.php(269): OC\Files\Cache\Scanner->scan('', true, 3)
nextcloud/server#7 /var/www/html/apps/files/lib/Command/Scan.php(164): OC\Files\Utils\Scanner->scan('/mario', true, NULL)
nextcloud/server#8 /var/www/html/apps/files/lib/Command/Scan.php(219): OCA\Files\Command\Scan->scanFiles('mario', '/mario', false, Object(Symfony\Component\Console\Output\ConsoleOutput), false, true, false)
nextcloud/server#9 /var/www/html/3rdparty/symfony/console/Command/Command.php(298): OCA\Files\Command\Scan->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#10 /var/www/html/core/Command/Base.php(177): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#11 /var/www/html/3rdparty/symfony/console/Application.php(1040): OC\Core\Command\Base->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#12 /var/www/html/3rdparty/symfony/console/Application.php(301): Symfony\Component\Console\Application->doRunCommand(Object(OCA\Files\Command\Scan), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#13 /var/www/html/3rdparty/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#14 /var/www/html/lib/private/Console/Application.php(211): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#15 /var/www/html/console.php(100): OC\Console\Application->run()
nextcloud/server#16 /var/www/html/occ(11): require_once('/var/www/html/c...')
nextcloud/server#17 {main}
```
It looks like the file is there twice, because of the two lines with `File	/mario/files/test.jpg` but I double checked, the file i there only once.

In the same installation I did the same before (copy some files to the data dir and add them via `scan`) - it always worked flawlessly (with ~16000 files) - till now. 
I checked with different files and various methods of copying over the files: 
* `rsync`
* mount the volume on my mac and copy a file in the finder directly
* mount the volume as external in another nextcloud instance and copy a file from that instance
it always results on the same error when I try to add the file via `files:scan`

I checked the file permissions as well, they are correct and same as for all other files in the nextcloud data dir.

Just in case I did run `files:cleanup` and `files:repair-tree` - did not help.

Uploading via the web interface and sync via e.g. the android client still works normally.

### Steps to reproduce

Duno, it worked before.

### Expected behavior

`scan` adds the files to nextcloud 

### Installation method

Official All-in-One appliance

### Nextcloud Server version

27

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

PostgreSQL

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "one-click-instance": true,
        "one-click-instance.user-limit": 100,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "check_data_directory_permissions": false,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "password": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "overwritehost": "octopus.jacksbox.de",
        "overwriteprotocol": "https",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "octopus.jacksbox.de"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "27.0.2.1",
        "overwrite.cli.url": "https:\/\/octopus.jacksbox.de\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "loglevel": "2",
        "log_type": "file",
        "logfile": "\/var\/www\/html\/data\/nextcloud.log",
        "log_rotate_size": "10485760",
        "log.condition": {
            "apps": [
                "admin_audit"
            ]
        },
        "preview_max_x": "2048",
        "preview_max_y": "2048",
        "jpeg_quality": "60",
        "enabledPreviewProviders": {
            "1": "OC\\Preview\\Image",
            "2": "OC\\Preview\\MarkDown",
            "3": "OC\\Preview\\MP3",
            "4": "OC\\Preview\\TXT",
            "5": "OC\\Preview\\OpenDocument",
            "6": "OC\\Preview\\Movie",
            "7": "OC\\Preview\\Krita",
            "0": "OC\\Preview\\Imaginary"
        },
        "enable_previews": true,
        "upgrade.disable-web": true,
        "mail_smtpmode": "smtp",
        "trashbin_retention_obligation": "auto, 30",
        "versions_retention_obligation": "auto, 30",
        "activity_expire_days": "30",
        "simpleSignUpLink.shown": false,
        "share_folder": "\/Shared",
        "memories.exiftool": "\/var\/www\/html\/custom_apps\/memories\/exiftool-bin\/exiftool-aarch64-musl",
        "memories.vod.path": "\/var\/www\/html\/custom_apps\/memories\/exiftool-bin\/go-vod-aarch64",
        "memories.vod.ffmpeg": "\/usr\/bin\/ffmpeg",
        "memories.vod.ffprobe": "\/usr\/bin\/ffprobe",
        "one-click-instance.link": "https:\/\/nextcloud.com\/all-in-one\/",
        "upgrade.cli-upgrade-link": "https:\/\/github.com\/nextcloud\/all-in-one\/discussions\/2726",
        "updatedirectory": "\/nc-updater",
        "davstorage.request_timeout": 3600,
        "htaccess.RewriteBase": "\/",
        "dbpersistent": true,
        "files_external_allow_create_new_local": false,
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "preview_imaginary_url": "http:\/\/nextcloud-aio-imaginary:9000",
        "memories.gis_type": 2,
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.19.0
  - admin_audit: 1.17.0
  - calendar: 4.4.4
  - camerarawpreviews: 0.8.2
  - circles: 27.0.1
  - cloud_federation_api: 1.10.0
  - comments: 1.17.0
  - contacts: 5.3.2
  - contactsinteraction: 1.8.0
  - dashboard: 7.7.0
  - dav: 1.27.0
  - deck: 1.10.0
  - federatedfilesharing: 1.17.0
  - federation: 1.17.0
  - files: 1.22.0
  - files_pdfviewer: 2.8.0
  - files_rightclick: 1.6.0
  - files_sharing: 1.19.0
  - files_trashbin: 1.17.0
  - files_versions: 1.20.0
  - firstrunwizard: 2.16.0
  - logreader: 2.12.0
  - lookup_server_connector: 1.15.0
  - memories: 5.2.1
  - nextcloud-aio: 0.4.0
  - nextcloud_announcements: 1.16.0
  - notes: 4.8.1
  - notifications: 2.15.0
  - notify_push: 0.6.3
  - oauth2: 1.15.1
  - password_policy: 1.17.0
  - photos: 2.3.0
  - previewgenerator: 5.3.0
  - privacy: 1.11.0
  - provisioning_api: 1.17.0
  - recognize: 4.3.1
  - recommendations: 1.6.0
  - related_resources: 1.2.0
  - serverinfo: 1.17.0
  - settings: 1.9.0
  - sharebymail: 1.17.0
  - support: 1.10.0
  - survey_client: 1.15.0
  - systemtags: 1.17.0
  - tasks: 0.15.0
  - text: 3.8.0
  - theming: 2.2.0
  - twofactor_backupcodes: 1.16.0
  - user_status: 1.7.0
  - viewer: 2.1.0
  - weather_status: 1.7.0
  - workflowengine: 2.9.0
Disabled:
  - bruteforcesettings: 2.7.0
  - encryption: 2.15.0
  - files_external: 1.19.0
  - suspicious_login: 5.0.0
  - twofactor_totp: 9.0.0
  - user_ldap: 1.17.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
empty
```

**Comment by Ra72xx** [comment]
This happens with a bare-metal Nextcloud installation, too. For me with my photo/video collection on local external storage and with PostgreSQL database. I have to deactivate the app, otherwise PostgreSQL spams my error logs with the errors above.

**Comment by raftersvk** [comment]
I am not sure that it's only a recognize bug, since I have the same problem by only using the "--generate-metadata" option

**Comment by jacksbox** [comment]
I can confirm that disabling recognise (4.3.1) solves this issue.
I did not try `"--generate-metadata"` though

**Comment by abrahamboray** [comment]
recognize  app is behind the BUG, deactivated and scan works fine now.

**Comment by raftersvk** [comment]
I come back again to the issue, since I think there is a larger problem.
I tried to scan again the same folder since some folders/files are missing (and I can't figure why because owner and permissions on those folders/files seems ok with the www-data user:group),
but here is the error ouput :
```
sudo docker exec -it -u www-data nextcloud-app php occ files:scan --path="/myuser/files/NAS/Photos" -vvv
Starting scan for user 1 out of 1 (myuser)
An unhandled exception has been thrown:
TypeError: OC\Files\Cache\Scanner::scanChildren(): Argument #6 ($oldSize) must be of type int, float given, called in /var/www/html/lib/private/Files/Cache/Scanner.php on line 347 and defined in /var/www/html/lib/private/Files/Cache/Scanner.php:391
Stack trace:
#0 /var/www/html/lib/private/Files/Cache/Scanner.php(347): OC\Files\Cache\Scanner->scanChildren('Photos', true, 3, 93977, true, 4519476975.0)
#1 /var/www/html/lib/private/Files/Utils/Scanner.php(269): OC\Files\Cache\Scanner->scan('Photos', true, 3)
#2 /var/www/html/apps/files/lib/Command/Scan.php(164): OC\Files\Utils\Scanner->scan('/myuser/files...', true, NULL)
#3 /var/www/html/apps/files/lib/Command/Scan.php(219): OCA\Files\Command\Scan->scanFiles('myuser', '/myuser/files...', false, Object(Symfony\Component\Console\Output\ConsoleOutput), false, true, false)
#4 /var/www/html/3rdparty/symfony/console/Command/Command.php(298): OCA\Files\Command\Scan->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#5 /var/www/html/core/Command/Base.php(177): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 /var/www/html/3rdparty/symfony/console/Application.php(1040): OC\Core\Command\Base->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 /var/www/html/3rdparty/symfony/console/Application.php(301): Symfony\Component\Console\Application->doRunCommand(Object(OCA\Files\Command\Scan), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 /var/www/html/3rdparty/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 /var/www/html/lib/private/Console/Application.php(211): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 /var/www/html/console.php(100): OC\Console\Application->run()
#11 /var/www/html/occ(11): require_once('/var/www/html/c...')

```

**Comment by marcelklehr** [comment]
@raftersvk that's a different issue

**Comment by marcelklehr** [comment]
I can't seem to reproduce this. It's true that recognize v4.3.1 adds a hook for files:scan, so it wouldn't be entirely unexpected that that would break something, but I've tried all v27 versions, even the upcoming v27.1 release and it just works: https://github.com/nextcloud/recognize/actions/runs/5904770847/job/16017548733

**Comment by marcelklehr** [comment]
My steps to reproduce are:

0. Install recognize and enable imagenet classification
1. Copy files to data/admin/files
2. Run `occ files:scan admin -vvv`

**Comment by raftersvk** [comment]
do you want us to debug something to help ?

**Comment by jacksbox** [comment]
could it be caused by:
1. nextcloud data dir is an external volume mounted on the host
2. sqlite vs. postgres (the tests above run in sqlite. Normally errors occur the other way round, pg works and sqlite not, but how know ;))
3. somehow influenced by the installed memories app as well (not sure if it's installed in the linked tests)

**Comment by raftersvk** [comment]
I can confirm that memories installed too on my NC

**Comment by marcelklehr** [comment]
@jacksbox Interesting points!

I can run the test with other databases, and I suggest you to try with the memories app disabled.

> nextcloud data dir is an external volume mounted on the host

Not sure about this. As the filesystem is transparent to nextcloud this shouldn't matter, but who knows. Anybody else with such a setup?

**Comment by Ra72xx** [comment]
Here: External local storage for my media files, "Memories" installed, too (only reason I tried out "Recognize" ;-) ).

**Comment by marcelklehr** [comment]
Tests with all databases are all passing: https://github.com/nextcloud/recognize/actions/runs/5906161003

**Comment by raftersvk** [comment]
same setup : external storage and memories installed

**Comment by marcelklehr** [comment]
I've just released https://github.com/nextcloud/recognize/releases/tag/v4.3.2 which tries to mitigate this bug by using proper hooks for the scan events rather than the hack I had before, please let me know if this fixes this issue. :hearts:

**Comment by Ra72xx** [comment]
Sorry, problem still there, lots of PostgreSQL after re-enabling the updated "recognize"-app :
`Aug 21 07:17:29 server postgresql[526618]: 2023-08-21 07:17:29.766 CEST [239623] DETAIL:  Key (objecttype, objectid, systemtagid)=(files, 11756525, 207) already exists.
Aug 21 07:17:29 server postgresql[526618]: 2023-08-21 07:17:29.766 CEST [239623] STATEMENT:  INSERT INTO "oc_systemtag_object_mapping" ("objectid", "objecttype", "systemtagid") VALUES($1, $2, $3`

**Comment by marcelklehr** [comment]
Mmh, this seems not directly related with recognize. Is there a stack trace? Have you tried occ files:scan again?

**Comment by Ra72xx** [comment]
files:scan seems to work without any problem. But as soon as I enable "recognize", PostgreSQL keeps throwing out this error. No error in Nextcloud-log, however. So maybe it's another error which occured at the same time in my installation.

**Comment by jacksbox** [comment]
`4.3.2` Solved the issue for me

**Comment by surapuramakhil** [comment]
this is occuring again

```bash
akhil@akhil-minipc:~/nextCloud$ sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ files:scan Akhil
Starting scan for user 1 out of 1 (Akhil)
Exception during scan: File entry could not be inserted but could also not be selected with getId() in order to perform an update. Please try again.
#0 /var/www/html/lib/private/Files/Cache/Scanner.php(284): OC\Files\Cache\Cache->insert('files/sandisk_s...', Array)
#1 /var/www/html/lib/private/Files/Cache/Scanner.php(217): OC\Files\Cache\Scanner->addToCache('files/sandisk_s...', Array, -1)
#2 /var/www/html/lib/private/Files/Cache/Scanner.php(501): OC\Files\Cache\Scanner->scanFile('files/sandisk_s...', 3, 1159566, false, true, Array)
#3 /var/www/html/lib/private/Files/Cache/Scanner.php(425): OC\Files\Cache\Scanner->handleChildren('files/sandisk_s...', true, 3, 1159566, true, 9136127, false)
#4 /var/www/html/lib/private/Files/Cache/Scanner.php(430): OC\Files\Cache\Scanner->scanChildren('files/sandisk_s...', true, 3, 1159566, true, -1, false)
#5 /var/www/html/lib/private/Files/Cache/Scanner.php(430): OC\Files\Cache\Scanner->scanChildren('files/sandisk_s...', true, 3, 490885, true, -1, 0)
#6 /var/www/html/lib/private/Files/Cache/Scanner.php(430): OC\Files\Cache\Scanner->scanChildren('files/sandisk_s...', true, 3, 421427, true, -1, 1)
#7 /var/www/html/lib/private/Files/Cache/Scanner.php(430): OC\Files\Cache\Scanner->scanChildren('files/sandisk_s...', true, 3, 490864, true, -1, 1)
#8 /var/www/html/lib/private/Files/Cache/Scanner.php(430): OC\Files\Cache\Scanner->scanChildren('files', true, 3, 409, true, 39488751930, 1)
#9 /var/www/html/lib/private/Files/Cache/Scanner.php(331): OC\Files\Cache\Scanner->scanChildren('', true, 3, 406, true, 39488751930)
#10 /var/www/html/lib/private/Files/Utils/Scanner.php(262): OC\Files\Cache\Scanner->scan('', true, 3)
#11 /var/www/html/apps/files/lib/Command/Scan.php(153): OC\Files\Utils\Scanner->scan('/Akhil', true, NULL)
#12 /var/www/html/apps/files/lib/Command/Scan.php(214): OCA\Files\Command\Scan->scanFiles('Akhil', '/Akhil', NULL, Object(Symfony\Component\Console\Output\ConsoleOutput), false, true, false)
#13 /var/www/html/3rdparty/symfony/console/Command/Command.php(326): OCA\Files\Command\Scan->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /var/www/html/core/Command/Base.php(161): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /var/www/html/3rdparty/symfony/console/Application.php(1078): OC\Core\Command\Base->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /var/www/html/3rdparty/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand(Object(OCA\Files\Command\Scan), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 /var/www/html/3rdparty/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /var/www/html/lib/private/Console/Application.php(183): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 /var/www/html/console.php(87): OC\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput))
#20 /var/www/html/occ(33): require_once('/var/www/html/c...')
#21 {main}
+---------+-------+-----+---------+---------+--------+--------------+
| Folders | Files | New | Updated | Removed | Errors | Elapsed time |
+---------+-------+-----+---------+---------+--------+--------------+
| 156     | 21291 | 0   | 9758    | 0       | 1      | 00:00:21     |
+---------+-------+-----+---------+---------+--------+--------------+
akhil@akhil-minipc:~/nextCloud$ ls

```

**Comment by marcelklehr** [comment]
@surapuramakhil Can you please open a new issue?

**Comment by surapuramakhil** [comment]
@marcelklehr I have done -vvv deleted problematic files, rerun it.

**Comment by marcelklehr** [comment]
And that fixed it?

**Comment by surapuramakhil** [comment]
ah you can consider as dodging hurdle
